### PR TITLE
Introduce AspectRatio struct

### LINKED ISSUE #10288: Inconsistent implementations for calculating aspect ratios
## Bevy version
- 0.11.3
- the main branch as of `b22db47`

## What went wrong
Aspect ratios should be calculated as `width / height`, not `height / width`.

In the code base of the version 0.11.3 and the main branch (`b22db47`), I found that `bevy_render/src/texture/image.rs` wrongly uses `height / width` while other files correctly use `width / height`.

### v0.11.3
https://github.com/bevyengine/bevy/blob/0d23d71c19c784ceb1acfbb134dda9ce0c2adc61/crates/bevy_render/src/texture/image.rs#L260-L263

### The main branch as of `b22db47`
https://github.com/bevyengine/bevy/blob/b22db47e1003db2d712baec46c0f0a1f88b0a837/crates/bevy_render/src/texture/image.rs#L542-L546

## Additional information
Other files use `width / height`.

https://github.com/bevyengine/bevy/blob/0d23d71c19c784ceb1acfbb134dda9ce0c2adc61/crates/bevy_ui/src/ui_node.rs#L388-L391

https://github.com/bevyengine/bevy/blob/0d23d71c19c784ceb1acfbb134dda9ce0c2adc61/crates/bevy_render/src/camera/projection.rs#L121-L127

https://github.com/bevyengine/bevy/blob/0d23d71c19c784ceb1acfbb134dda9ce0c2adc61/crates/bevy_pbr/src/light.rs#L714

https://github.com/bevyengine/bevy/blob/0d23d71c19c784ceb1acfbb134dda9ce0c2adc61/crates/bevy_core_pipeline/src/bloom/settings.rs#L214

## Idea
Calculations related to aspect ratio can be confusing. So, what about the idea of creating a newtype struct like `pub struct AspectRatio(f32)`?

### PR DESCRIPTION
**Pull request by Olle-Lukowski**
# Objective

- Fix an inconsistency in the calculation of aspect ratio's. 
- Fixes #10288 

## Solution

- Created an intermediate `AspectRatio` struct, as suggested in the issue. This is currently just used in any places where aspect ratio calculations happen, to prevent doing it wrong. In my and @mamekoro 's opinion, it would be better if this was used instead of a normal `f32` in various places, but I didn't want to make too many changes to begin with.

## Migration Guide
- Anywhere where you are currently expecting a f32 when getting aspect ratios, you will now receive a `AspectRatio` struct. this still holds the same value.

### CODE REVIEW

File: crates/bevy_render/src/texture/image.rs [hunk 1]
```diff
@@ -539,10 +539,10 @@ impl Image {
         self.texture_descriptor.size.height
     }
 
-    /// Returns the aspect ratio (height/width) of a 2D image.
+    /// Returns the aspect ratio (width / height) of a 2D image.
     #[inline]
     pub fn aspect_ratio(&self) -> f32 {
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/texture/image.rs:544]: I feel like this should just return an `AspectRatio`.

File: crates/bevy_math/src/aspect_ratio.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+impl AspectRatio {
+    /// Create a new `AspectRatio` from a given `width` and `height`.
+    #[inline]
+    pub fn new(width: f32, height: f32) -> Self {
+        Self(width / height)
+    }
+}
+
+impl From<(f32, f32)> for AspectRatio {
```
Comment (alice-i-cecile) [inline crates/bevy_math/src/aspect_ratio.rs:14]: I think we should not include this: it's easy to make exactly the same mistake.

File: crates/bevy_render/src/texture/image.rs (hunk 1 above)
Comment (Olle-Lukowski) [reply crates/bevy_render/src/texture/image.rs:544]: Alright, will get to it straight away.

File: crates/bevy_pbr/src/light.rs [hunk 3]
```diff
@@ -719,7 +721,8 @@ impl ClusterConfig {
             ClusterConfig::FixedZ {
                 total, z_slices, ..
             } => {
-                let aspect_ratio = screen_size.x as f32 / screen_size.y as f32;
+                let aspect_ratio: f32 =
+                    AspectRatio::new(screen_size.x as f32, screen_size.y as f32).into();
```
Comment (alice-i-cecile) [inline crates/bevy_pbr/src/light.rs:725]: IMO we should have a `from_pixels` method or something that takes the integer values.

Comment (Olle-Lukowski) [reply crates/bevy_pbr/src/light.rs:725]: You mean in the `AspectRatio`?

Comment (alice-i-cecile) [reply crates/bevy_pbr/src/light.rs]: Yeah, an impl on this would be useful.

Comment (Olle-Lukowski) [reply crates/bevy_pbr/src/light.rs]: Thought so, it's added in the latest commit.

File: crates/bevy_render/src/texture/image.rs [hunk 4]
```diff
@@ -539,10 +539,10 @@ impl Image {
         self.texture_descriptor.size.height
     }
 
-    /// Returns the aspect ratio (height/width) of a 2D image.
+    /// Returns the aspect ratio (width / height) of a 2D image.
     #[inline]
-    pub fn aspect_ratio(&self) -> f32 {
-        self.height() as f32 / self.width() as f32
+    pub fn aspect_ratio(&self) -> AspectRatio {
+        AspectRatio::new(self.width() as f32, self.height() as f32)
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/texture/image.rs:545]:
```suggestion
        AspectRatio::from_pixels(self.width(), self.height())
```

Comment (Olle-Lukowski) [reply crates/bevy_render/src/texture/image.rs:545]: seems like I forgot something

**Review by alice-i-cecile — APPROVED** [review]

**Comment by Olle-Lukowski** [comment]
Anything I need to do before this can be merged?

**Comment by alice-i-cecile** [comment]
Other than resolving merge conflicts, probably not. We need a second reviewer; let me ask around for one.

**Review by 100-TomatoJuice — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Merging! FYI, next time you probably want to create a branch with a unique name on your repo for each PR: it really helps you keep things straight.

**Pull request merged**